### ISSUE TITLE
GKE cluster goes to updating after edit with no changes

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#18853]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Create GKE cluster with default values.
2. Wait for GKE cluster to fully provision.
3. Edit cluster, do not change any values, and save.

**Result:**
Cluster goes to updating in UI and provisions in backend

**Other details that may be helpful:**
Provisioning is triggered because when the comparison for spec config and applied spec config fails. It appears that values, such as minNodeCount and maxNodeCount are being dropped

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):
Master 3/12

**Comment by loganhz** [comment]
@fyery-chen @n313893254 Can you confirm if this issue can get fixed in https://github.com/rancher/ui/pull/2753?

**Comment by loganhz** [comment]
We can confirm Wes's PR works

**Comment by davidnuzik** [comment]
Version: Master 2.2 (3/13/19) e090397b4c72
Confirmed issue.
When I edit my GKE cluster (but do not make any changes such as to nodecount or node pool autoscaling which are only editable options as of v2.2) and save the cluster does NOT update any longer.

I see we make a PUT to update but this doesn't cause anything to happen. The cluster does not update as expected.

I smoke tested this area as well, try enabling autoscaling, increase node count, etc to ensure this is still working correctly. I did not encounter any issues. When I did make these changes the cluster DID go into an updating state, as expected. I confirmed the changed took effect via gcloud web console.
